packages/contracts/contracts/tcr/AddressRegistry.sol

```diff
@@ -37,14 +37,14 @@ contract AddressRegistry {
     bool resolved;          // Indication of if challenge is resolved
     uint stake;             // Number of tokens at stake for either party during challenge
     uint totalTokens;       // (remaining) Number of tokens used in voting by the winning side
-    mapping(address => bool) tokenClaims; // Indicates whether a voter has claimed a reward yet
+    mapping(address => bool) hasClaimedTokens; // Indicates whether a voter has claimed a reward yet
   }
 
   // Maps challengeIDs to associated challenge data
-  mapping(uint => Challenge) internal challenges;
+  mapping(uint => Challenge) public challenges;
```

Comment:
Undocumented behaviour use here. Additionally with the Challenge being big and having a mapping inside, it sounds wonky. Does it work?

We could make use of structs long term by supporting abiv2 but it's also a bit of work

Reply:
the mapping inside the Challenge struct is the only part that's not exposed publicly. When you do `registry.challenges(challengeID)` it will return an array with everything but the `hasClaimedTokens` mapping, so we still need the getter function for that.